src/test/ui/error-codes/E0423.stderr

```diff
@@ -3,6 +3,19 @@ error: expected type, found `1`
    |
 LL |     if let S { x: _x, y: 2 } = S { x: 1, y: 2 } { println!("Ok"); }
    |                                       ^ expecting a type here because of type ascription
+   |
+   = note: type ascription is a nightly only feature that lets you annotate expressions with a type: `<expr>: <type>`
+note: this expression is annotated with type ascription...
+  --> $DIR/E0423.rs:12:36
+   |
+LL |     if let S { x: _x, y: 2 } = S { x: 1, y: 2 } { println!("Ok"); }
```

Comment:
The message at the moment is quite long. I wonder if we could collapse it into a single snippet. Also, I think the colon could come first, as it is that which triggers the ascription.
```
note: this expression expects an ascribed type
  --> $DIR/E0423.rs:12:36
   |
LL |     if let S { x: _x, y: 2 } = S { x: 1, y: 2 } { println!("Ok"); }
   |                                     - the colon here...
   |                                    ^ ...implies that this expression is ascribed a type
   = help: this might be indicative of a syntax error elsewhere
```

Reply:
@varkor the problem is that we can't add them as span_labels in the main error without it getting too crowded, and secondary diagnostic windows do not let you (at the moment) specify span labels. I thought that although it is unnecessarily verbose, it is better than the alternative. Once we have more flexibility to actually accomplish the output you suggest, I would be happy to implement it.

Reply:
An option would be

```
note: this expression expects an ascribed type after the colon
  --> $DIR/E0423.rs:12:36
   |
LL |     if let S { x: _x, y: 2 } = S { x: 1, y: 2 } { println!("Ok"); }
   |                                    ^
   = help: this might be indicative of a syntax error elsewhere
```

Reply:
@estebank: ahh, I see.

I prefer the suggested option in the interim: I think it captures all the necessary information, without being too verbose.

Reply:
Changed.